starter_code/src/movies.js

```diff
@@ -1,15 +1,373 @@
+function orderAlphabetically(arrNotMutated) {
+  arrTemp = [...arrNotMutated];
+  if (arrTemp.length == 0) {
+    return [null];
+  } else {
+    arrTemp.sort(function(a, b) {
+      return (a.title > b.title) - 0.5;
+    });
+
+    arrTemp.length >= 20 ? (arrTemp = arrTemp.slice(0, 20)) : arrTemp;
+
+    var arrFinal = [];
+
+    arrTemp.forEach(v => arrFinal.push(v.title));
+
+    return arrFinal;
+  }
+}
 
 // Iteration 4: All rates average - Get the average of all rates with 2 decimals
+function ratesAverage(arr) {
+  if (arr.length == 0) {
+    return 0;
+  } else {
+    sum = 0;
+    arr.forEach(v => (!isNaN(v.rate) ? (sum += v.rate) : 1));
+    return Math.round((sum / arr.length) * 100) / 100;
+  }
+}
+
+// console.log (ratesAverage(([{ rate: 8 }, { rate: 8 }])));
 
 // Iteration 5: Drama movies - Get the average of Drama Movies
+function dramaMoviesRate(arr) {
+  if (arr.length == 0 || arr.filter(v => v.genre == "Drama").length == 0) {
+    return 0;
+  } else {
+    var sum = 0;
+    var count = 0;
+    arr
+      .filter(v => v.genre == "Drama")
+      .forEach(v => {
+        if (!isNaN(v.rate)) {
+          sum += v.rate;
+          count++;
+        }
+      });
+    return Math.round((sum / count) * 100) / 100;
+  }
+}
 
 // Iteration 6: Time Format - Turn duration of the movies from hours to minutes
 
+function durationConvert(st) {
+  hIndex = st.split("").indexOf("h");
+  mIndex = st.split("").indexOf("m");
+
+  if (mIndex == -1) {
+    st = Number(st.slice(0, hIndex)) * 60;
+  } else {
+    if (hIndex == -1) {
+      st = Number(st.slice(hIndex + 1, mIndex));
+    } else {
+      st =
+        Number(st.slice(0, hIndex)) * 60 + Number(st.slice(hIndex + 1, mIndex));
+    }
+  }
+  return st;
+}
+
+function turnHoursToMinutes(arrTemp) {
+  const arr = [...arrTemp];
+  if (arr.length === 0) {
+    return null;
+  } else {
+    const arr2 = arr.map(v => {
+      let newObj = {...v};
+      newObj.duration=durationConvert(newObj.duration);
+      return newObj;
+    });
+    return arr2;
+  }
+}
+
+console.log(typeof turnHoursToMinutes(arr1)[0].duration);
+
 // BONUS Iteration: Best yearly rate average - Best yearly rate average
+
+function bestYearAvg(arr) {
+  if (arr.length == 0) {
+    return null;
+  } else {
+    var years = [];
+
+    arr.forEach(v => {
+      if (years.indexOf(v.year) == -1) {
+        years.push(v.year);
+      }
+    });
+
+    years.sort();
+
+    max = 0;
+
+    var averages = new Array(years.length);
+    var count = new Array(years.length);
+
+    years.forEach((v, i) => {
+      arr2 = arr.filter(element => element.year == years[i]);
+      averages[i] = arr2.reduce((acc, v2) => acc + v2.rate, 0);
+      count[i] = arr2.length;
+    });
+
+    averages.forEach((v, i, arr) => (arr[i] = v / count[i]));
+
+    daAverage = Math.max(...averages);
+    daYear = years[averages.indexOf(daAverage)];
+
+    return `The best year was ${daYear} with an average rate of ${daAverage}`;
+  }
+}
+
```

Comment:
Nice solution.

Just be careful about O(n), you are iterating many times over arrays.
Some more information about what is O(n)

https://en.wikipedia.org/wiki/Big_O_notation